src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -40,17 +43,48 @@ class DataSource(graphFactory: OrientGraphFactory) extends LazyLogging {
       rollbackOnly.set(false)
       graph.begin()
       val result = f(new RawlsTransaction(graph, this))
+      completeTransaction(graph)
+      result
+    } catch {
+      case t: Throwable =>
+        graph.rollback()
+        throw t
+    } finally {
+      graph.shutdown()
+    }
+  }
+
```

Comment:
the following function completes or rolls back the transaction in a future at the end of the result future